superset/sql_parse.py

```diff
@@ -89,8 +89,8 @@ def __process_identifier(self, identifier):
         # exclude subselects
         if '(' not in str(identifier):
             table_name = self.__get_full_name(identifier)
-            if not table_name.startswith(CTE_PREFIX):
-                self._table_names.add(self.__get_full_name(identifier))
+            if table_name and not table_name.startswith(CTE_PREFIX):
+                self._table_names.add(table_name)
```

Comment:
Note there's no need to recall `self.__get_full_name(identifier)` again as the return value is captured by the `table_name` variable.